pkg/apis/autoscaling/types.go

```diff
@@ -92,11 +92,73 @@ type HorizontalPodAutoscalerSpec struct {
 	// more information about how each type of metric must respond.
 	// +optional
 	Metrics []MetricSpec
+	// constraints contain HPA object configuration parameters
+	// that configure each particular HPA scaling behaviour.
+	// +optional
+	Constraints *HPAScaleConstraints
 }
 
 // MetricSourceType indicates the type of metric.
 type MetricSourceType string
 
+// HPAScaleConstraints configures a scaling velocity for Up and Down direction
+// (scaleUp and scaleDown fields respectively)
+type HPAScaleConstraints struct {
+	// constraint value for scaling Up
+	// +optional
+	ScaleUp *HPAScaleConstraintValue
+	// constraint value for scaling Down
+	// +optional
+	ScaleDown *HPAScaleConstraintValue
+}
+
+// HPAScaleConstraintRateValue configures the scaling velocity
+// by specifying the "absolute" value (in number of pods) and "relative" values (in percents)
+// and the corresponding time period, for which this constraints are applied
+// If both parameters are specified the largest constraint is used.
+type HPAScaleConstraintRateValue struct {
+	// In order to enable the constraint, either `pods` or `percent` should be set
+	// To support backward compatibility it's possible that none may be set,
+	//   in which case the default values are used
+
+	// Pods specifies the absolute speed, in number of pods per <PeriodSeconds> time period
+	// i.e. if Pods = 5 , then we can add/remove 5 pods (10 -> 15 or 10 -> 5)
+	//      if Pods = -1 (or any negative value), this parameter should not be used
+	//		if Pods is not specified, the default value is used:
+	//			for scaleUp: Pods = 4
+	//			for scaleDown: Pods = -1
+	// more info could be found in the KEP https://github.com/kubernetes/enhancements/pull/883
+	// +optional
+	Pods *int32
```

Comment:
I wonder why the fields are pointers .
A nil pointer can be represented by -1 if primitive type is used.